module1/exercises/07_pararel_countif.cpp

```diff
@@ -0,0 +1,78 @@
+#include <iostream>
+#include <thread>
+#include <numeric>
+#include <algorithm>
+#include <vector>
+#include <chrono>
+#include <functional>
+
+constexpr size_t minSize = 1'000'000;
+
+bool IsOdd (int i) { return ((i%2)==1); }
+
+template <typename IT, class UnaryPredicate>
+long long parallelCountIf(IT first, IT last, UnaryPredicate pred) {
+    size_t size = std::distance(first, last);
+    if (size < minSize)
+        return std::count_if(first, last, pred);
+
+    unsigned int numberOfThreads = std::thread::hardware_concurrency();
+    if( ! (numberOfThreads > 0 && numberOfThreads < 50)) {
+        numberOfThreads = 2;
+    }
+
+    std::cout << "Will split work to " << numberOfThreads << " threads." << std::endl;
+
+    std::vector<long long> results(numberOfThreads-1);
+    std::vector<std::thread> threads;
+
+    auto chunkSize = size / numberOfThreads;  // range for single thread
+    auto localFirst = first;
+    auto localLast = std::next(localFirst, chunkSize);
+    std::cout << "Split " << size << " into subranges with size of " << chunkSize << std::endl;
+
+    for (unsigned int i = 0; i < numberOfThreads-1; i++)
+    {
+        localLast = std::next(localFirst, chunkSize);
+        auto lambda = [&results, i, &pred](IT first, IT last){
+            auto result = std::count_if(first, last, pred);
+            results[i] = result; };
+        std::thread t{ lambda, localFirst, localLast };
+        threads.emplace_back(std::move(t));
```

Comment:
You can also merge it like this
```suggestion
        threads.emplace_back(lambda, localFirst, localLast);
```